Migration of Process Module

harvey0100 opened the pull request:
Migration of Process Module
Improved Unit/Functional / RST Docstrings

Reference: https://github.com/avocado-framework/aautils/issues/52

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a feature-rich subprocess and command-execution toolkit (process inspection, safe termination/tree operations, streaming output, structured command results, and helpers).

* **Tests**
  * Added extensive unit and functional tests covering subprocess lifecycle, timeouts, encoding, signaling, environment propagation, and edge cases.
  * Added test utilities to configure loggers and decorators to skip tests by runtime level or missing paths.

* **Documentation**
  * Documented the new Process utilities in the developer docs.

* **Chores**
  * Added module metadata describing the new process utilities.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

harvey0100 committed:
Migration of Process Module

Migration of Process Module
Improved Unit/Functional / RST Docstrings

Reference: https://github.com/avocado-framework/aautils/issues/52
Signed-off-by: Harvey Lynden <hlynden@redhat.com>

richtja reviewed (changes requested):
Hi @harvey0100, can you please fix the CI issues? It looks like that is related to some wrong imports. 

Another thing is that IMO the `process` utility can under the `devel` module instead of its own.

Also, there are some differences between this and avocado version:

```
diff autils/process/process.py ../avocado/avocado/utils/process.py
33,35c33,34
< from autils.devel import astring
< from autils.devel.wait import wait_for
< from autils.file import path
---
> from avocado.utils import astring, path
> from avocado.utils.wait import wait_for
98c97
<         if system_output("id -u", ignore_status=True, sudo=True).strip() == "0":
---
>         if getoutput("id -u", sudo=True).strip() == "0":
``` 

And last part, I am missing the documentation here.

harvey0100 committed:
tests: fix test_can_sudo_oserror mocking wrong function

The test was mocking 'system' but can_sudo() without a cmd
argument calls getoutput() internally.

harvey0100 commented:
Now includes the latest CI fixes from avocado side.

richtja reviewed (changes requested):
Hi @harvey0100, unfortunately there are still differences between avocado and this. Can you please look into that?

```
98c97
<         if system_output("id -u", ignore_status=True, sudo=True).strip() == "0":
---
>         if getoutput("id -u", sudo=True).strip() == "0":

```
